packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts

```diff
@@ -17,60 +17,58 @@ import {
-    return (
-        isFunction(listener) ||
-        (!isNull(listener) && isObject(listener) && isFunction(listener.handleEvent))
-    );
-}
-
-function addEventListener(
+function patchedAddEventListener(
     this: EventTarget,
-    type: string,
-    listener: EventListenerOrEventListenerObject,
-    optionsOrCapture?: boolean | AddEventListenerOptions
+    _type: string,
+    _listener: EventListenerOrEventListenerObject,
+    _optionsOrCapture?: boolean | AddEventListenerOptions
 ) {
-    // TODO [#2134]: Delegate this validation to the browser instead of silently swallowing
-    if (!isValidEventListener(listener)) {
-        return;
-    }
+    const args = [...arguments];
     if (isHostElement(this)) {
-        addCustomElementEventListener(this, type, listener);
-    } else {
-        const wrapperFn = getEventListenerWrapper(listener);
-        nativeAddEventListener.call(this, type, wrapperFn, optionsOrCapture);
+        // Typescript does not like it when you treat the `arguments` object as an array
+        // @ts-ignore type-mismatch
+        return addCustomElementEventListener(this, ...args);
     }
+    if (args.length > 1) {
+        args[1] = getEventListenerWrapper(args[1]);
+    }
+    // Typescript does not like it when you treat the `arguments` object as an array
+    // @ts-ignore type-mismatch
+    return nativeAddEventListener.apply(this, args);
 }
 
-function removeEventListener(
+function patchedRemoveEventListener(
     this: EventTarget,
-    type: string,
-    listener: EventListenerOrEventListenerObject,
-    optionsOrCapture?: boolean | EventListenerOptions
+    _type: string,
+    _listener: EventListenerOrEventListenerObject,
+    _optionsOrCapture?: boolean | EventListenerOptions
 ) {
-    // TODO [#2134]: Delegate this validation to the browser instead of silently swallowing
-    if (!isValidEventListener(listener)) {
-        return;
-    }
+    const args = [...arguments];
     if (isHostElement(this)) {
-        removeCustomElementEventListener(this, type, listener);
-    } else {
-        const wrapperFn = getEventListenerWrapper(listener);
-        nativeRemoveEventListener.call(this, type, wrapperFn, optionsOrCapture);
-        // Account for listeners that were added before this polyfill was applied
-        nativeRemoveEventListener.call(this, type, listener, optionsOrCapture);
+        // Typescript does not like it when you treat the `arguments` object as an array
+        // @ts-ignore type-mismatch
+        return removeCustomElementEventListener(this, ...args);
+    }
+    if (arguments.length > 1) {
+        args[1] = getEventListenerWrapper(args[1]);
     }
+    // Typescript does not like it when you treat the `arguments` object as an array
+    // @ts-ignore type-mismatch
+    nativeRemoveEventListener.apply(this, args);
+    // Account for listeners that were added before this polyfill was applied
+    // @ts-ignore type-mismatch
+    nativeRemoveEventListener.apply(this, arguments);
```

Comment:
```suggestion
    nativeRemoveEventListener.apply(this, args);
```

Reply:
`args[1]` is the wrapped listener. We want `arguments` so that we can remove the ones that were added earlier.

Reply:
I can pass `[...arguments]` if that's what you are suggesting.

Reply:
It's fine as-is.